src/app/Marine2/components/ui/Box/Box.tsx

```diff
@@ -49,7 +70,17 @@ const Box = ({ children, icon, title, className, linkedView, getBoxSizeCallback
           </div>
         )}
       </div>
-      <div className={"w-full min-h-0 h-full pt-2"}>{children}</div>
+      <div className={"w-full min-h-0 h-full pt-2"}>
+        {withPagination && paginationOrientation === "horizontal" && (
+          <Paginator pageSelectorPropsSetter={setPageSelectorProps}>{children}</Paginator>
+        )}
+        {withPagination && paginationOrientation === "vertical" && (
+          <Paginator orientation={"vertical"} selectorLocation={"right-full"}>
```

Comment:
Why don't we use `pageSelectorPropsSetter={setPageSelectorProps}` prop if the case of `paginationOrientation === "vertical"`?

Reply:
If the pagination needs to be vertical then the ```PageSelector``` does not need to be a separate component. If ```pageSelectorPropsSetter``` is not given to ```Paginator```, it will use its own PageSelector component based on selectorLocation.

Reply:
Ah, I see now. I was confused by the `props` part in the name `setPageSelectorProps`. So it's not the props setter, it's the external PageSelector component, right? Would it be better to call it like just `pageSelector`?

Reply:
it is a props setter because we are not passing the ```PageSelector``` component. When it is specified separately as a sibling rather than a child, the component that is the parent of both of them has to ensure the possibility of transferring the state between the two so the parent has to have its own ```PageSelectorProps``` state that both of these components use, so I thought it makes sense